resource/template/resource_test.go

```diff
@@ -40,6 +43,27 @@ keys = [
 ]
 `
 
+// getRandomGroup randomly selects a gid from the Groups the current user is in
+func getRandomGroup() (gid int, err error) {
+	ogid := os.Getgid()
+	gid = ogid
+
+	// pick a random group and set the file to that group
+	groups, err := os.Getgroups()
+	if err != nil {
+		return
+	}
+
+	// It is possible this never terminates assert the length is greater than 1
```

Comment:
how do you suppose we do that ? non root group that is always on any possible system ?